R-package/configure.ac

```diff
@@ -0,0 +1,167 @@
+### configure.ac                    -*- Autoconf -*-
+# Template used by Autoconf to generate 'configure' script. For more see:
+#   * https://unconj.ca/blog/an-autoconf-primer-for-r-package-authors.html
+#   * https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Configure-and-cleanup
+
+AC_PREREQ(2.69)
+AC_INIT([lightgbm], [2.3.2], [], [lightgbm], [])
+
+###########################
+# find compiler and flags #
+###########################
+
+AC_MSG_CHECKING([location of R])
+AC_MSG_RESULT([${R_HOME}])
+
+# set up CPP flags
+# find the compiler and compiler flags used by R.
+: ${R_HOME=`R HOME`}
+if test -z "${R_HOME}"; then
+    echo "could not determine R_HOME"
+    exit 1
+fi
+CC=`"${R_HOME}/bin/R" CMD config CC`
+CXX=`"${R_HOME}/bin/R" CMD config CXX11`
+CFLAGS=`"${R_HOME}/bin/R" CMD config CFLAGS`
+CPPFLAGS=`"${R_HOME}/bin/R" CMD config CPPFLAGS`
+
+################
+# LGB_CPPFLAGS #
+################
+
+LGB_CPPFLAGS=""
+
+AC_MSG_CHECKING([whether using R version 3.5.0 or later])
+ac_pkg_r35=no
+R_35_OR_ABOVE=$(
+    ${R_HOME}/bin/Rscript -e "
+        R_ver <- as.double(R.version[['major']]) + as.double(R.version[['minor']]) / 10.0;
+        cat(as.character(R_ver > 3.5))
+    "
+)
+if test ${R_35_OR_ABOVE} = "TRUE"
+then
+    ac_pkg_r35=yes
+    LGB_CPPFLAGS+=" -DR_VER_ABOVE_35"
+fi
+AC_MSG_RESULT([${ac_pkg_r35}])
+
+# Compile for GPU?
+AC_ARG_ENABLE(
+    [gpu],
+    AS_HELP_STRING(
+        [--enable-gpu],
+        [Pass this flag to install the R package for usage with GPU]
+    ),
+    [
+        LGB_CPPFLAGS+=" -DUSE_GPU"
+    ]
+)
```

Comment:
note to self: remove this and move it to separate issue

Reply:
moved to #3206 and removed from this PR in https://github.com/microsoft/LightGBM/pull/3188/commits/ebe1d4789f49f1c16b30f211fe2acf379d607e45